pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -239,11 +240,16 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) Results {
 			delete(errors, k)
 		}
 	}
-	return Results{
+	results := Results{
 		NewNodeClaims: s.newNodeClaims,
 		ExistingNodes: s.existingNodes,
 		PodErrors:     errors,
 	}
+	results.TruncateInstanceTypes(MaxInstanceTypes)
+	UnschedulablePodsCount.With(
```

Comment:
The reason I did not pull it out is because in provisioner I did not have access to the scheduler ID. This was a point of discussion because it seemed I could either pull it out and not have SchedulingIDLabel or I would need to change the existing interface.